client/my-sites/hosting/staging-site-card/index.js

```diff
@@ -157,6 +152,38 @@ export const StagingSiteCard = ( { currentUserId, disabled, siteId, siteOwnerId,
 		},
 	} );
 
+	const isTrasferInProgress =
+		addingStagingSite ||
+		( showManageStagingSite &&
+			! isStagingSiteTransferComplete &&
+			( transferStatus !== null || wasCreating ) );
+
+	useEffect( () => {
+		// We know that a user has been navigated to an other page and came back if
+		// The transfer status is not in a final state (complete or failure)
+		// the staging site exists
+		// the site is not reverting
+		// the user owns the staging site
+		// and wasCreating that is set up by the add staging site button is false
+		if (
+			! wasCreating &&
+			! isStagingSiteTransferComplete &&
+			stagingSite.id &&
+			transferStatus !== transferStates.REVERTED &&
+			hasSiteAccess &&
+			! isReverting
+		) {
+			setWasCreating( true );
+		}
```

Comment:
Here is a small edge case where we are displaying the success notice.

The way to reproduce is by clearing the calypso_sotre indexedDB and refreshing the page.

I think this error comes from receiving the staging site but in the redux store we don't have its status yet. So the transfer status is null until the request finishes.

I'm fine deploying this PR and creating an issue for that.

## Screencast

https://user-images.githubusercontent.com/779993/235206011-064bd478-c014-4d49-a28b-a5b918a2ffb4.mp4

## State:

![Screenshot 2023-04-28 at 17 36 13](https://user-images.githubusercontent.com/779993/235206983-a885fd76-c075-4a69-b98a-6ec225130dbd.png)

Not sure if we really need this useEffect hook.

Reply:
@sejas we need, as `wasCreating` is used to set that the site is being created. I'm sure that we can simplify that, however, I think we need to revisit, and probably rewrite this component.
I don't mind filling a new issue for that. I find that issue, not a big deal, in comparison to what is happening right now when the user is moving back and forth to the pages.